sky/packages/sky/lib/src/gestures/arena.dart

```diff
@@ -72,6 +73,32 @@ class GestureArena {
     _tryToResolveArena(key, state);
   }
 
+  /// Force resolution on this arena, giving the win to the first member
+  void sweep(Object key) {
+    _GestureArenaState state = _arenas[key];
+    if (state == null)
+      return;  // This arena either never existed or has been resolved.
+    assert(!state.isOpen);
+    if (state.isHeld)
+      return;  // This arena is being held for a long-lived member
+    if (!state.members.isEmpty) {
+      // First member wins
+      state.members.first.acceptGesture(key);
+      // Give all the other members the bad news
+      for (int i = 1; i < state.members.length; i++)
+        state.members[i].rejectGesture(key);
+    }
+    _arenas.remove(key);
+  }
+
+  /// Prevent the arena from being swept
+  void hold(Object key) {
```

Comment:
I think we also need to handle the case where a sweep is pending upon release, separately from the case where a hold and release are done before a sweep would normally occur. So that would add another piece of state.